contracts/src/Verifier.sol

```diff
@@ -73,101 +119,289 @@ contract Verifier is IVerifier, Ownable {
         });
-        contentIdToModel[modelContentId] = _model;
+        contentIdToModel[_modelContentId] = _model;
 
-        emit ModelRegistered(modelContentId, msg.sender);
+        emit ModelRegistered(_modelContentId, msg.sender);
 
         return _model;
     }
 
     function getModel(
-        Hash modelContentId
+        Hash _modelContentId
+    ) external view checkIfModelExists(_modelContentId) returns (Model memory) {
+        return contentIdToModel[_modelContentId];
+    }
+
+    function getModels(
+        uint32 _offset,
+        uint32 _limit
     )
         external
         view
-        checkIfModelExists(modelContentId)
-        returns (Model memory model)
+        validateOffsetParameter(_offset, models.length)
+        validateLimitParameter(_limit)
+        returns (ModelArrayElement[] memory)
     {
-        return contentIdToModel[modelContentId];
+        return _paginateModels(models, _offset, _limit);
     }
 
-    function getModels(
-        uint32 offset,
-        uint32 limit
+    function getModelsByOwnerAddress(
+        address _ownerAddress,
+        uint32 _offset,
+        uint32 _limit
     )
         external
         view
-        validateLimitParameter(limit)
-        returns (ModelArrayElement[] memory paginatedModels)
+        checkIfModelOwnerExists(_ownerAddress)
+        validateOffsetParameter(
+            _offset,
+            ownerAddressToModels[_ownerAddress].length
+        )
+        validateLimitParameter(_limit)
+        returns (ModelArrayElement[] memory)
     {
-        require(
-            offset < models.length,
-            "offset must be < length of list of models"
-        );
+        return
+            _paginateModels(
+                ownerAddressToModels[_ownerAddress],
+                _offset,
+                _limit
+            );
+    }
 
-        return _paginateModels(models, offset, limit);
+    function updateModel(
+        Hash _modelContentId,
+        string calldata _modelName,
+        string calldata _modelDescription
+    )
+        external
+        checkIfModelExists(_modelContentId)
+        isModelOwner(_modelContentId)
+        validateModelParameters(_modelName, _modelDescription)
+    {
+        contentIdToModel[_modelContentId].name = _modelName;
+        contentIdToModel[_modelContentId].description = _modelDescription;
     }
 
-    function getModelsByOwnerAddress(
-        address ownerAddress,
-        uint32 offset,
-        uint32 limit
+    function disableModel(
+        Hash _modelContentId
+    )
+        external
+        checkIfModelExists(_modelContentId)
+        isModelOwner(_modelContentId)
+    {
+        contentIdToModel[_modelContentId].isDisabled = true;
+    }
+
+    function commit(
+        Hash _modelContentId,
+        Hash _merkleRoot
+    ) external checkIfModelExists(_modelContentId) returns (Hash, Hash, uint8) {
+        Hash _commitId = _generateCommitId(_modelContentId, _merkleRoot);
```

Comment:
Not really. This value is generated as a combination of model (ID) AND test data (Merkle tree).

Reply:
Remove `merkleRoot` from `Commit` struct.

Reply:
> Remove merkleRoot from Commit struct.

Actually, it shouldn't be removed because [`multiProofVerifyCalldata`](https://docs.openzeppelin.com/contracts/4.x/api/utils#MerkleProof-multiProofVerifyCalldata-bytes32---bool---bytes32-bytes32---) uses merkleRoot to verify the given proofs.